packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -90,6 +90,8 @@ export default class ScriptTransformer {
     fileData: string,
     filename: Config.Path,
     instrument: boolean,
+    supportsDynamicImport: boolean,
```

Comment:
I made them non-optional for all private methods and default `false` for the public ones

Reply:
should ideally use options objects, as 2-3 boolean flag inputs are quite confusing

Reply:
Generally agree, but IIRC functions with more arguments are optimized better than the ones with object arguments, and this is kinda hot path

Reply:
potentially we could add separate functions? the es module path can be  async (or rather, always is) as well, so that might tie nicely into #9504. Then we could ditch the `supportsDynamicImport` option since that works for both esm and cjs (and can be async) and just vary on the static version